lib/ddtrace/contrib/qless/tracer_cleaner.rb

```diff
@@ -15,10 +15,7 @@ def around_perform(job)
         private
 
         def forked?
-          pin = Datadog::Pin.get_from(::Qless)
-          return false unless pin
-
-          pin.config[:forked] == true
+          Datadog::Tracing.configuration_for(::Qless, :forked) == true
         end
```

Comment:
⬆️  Wait, we're checking if Qless forked and use that to do `Tracing.shutdown!` (in line 11). Does it mean that tracing will be broken for the forked instance? Isn't `shutdown!` assumed to be irreversible?

(Same note for Resque)

Reply:
Presumably the `shutdown!` happens at the end of the fork, such that it's irrelevant whether tracing stops working. The point of doing so (presumably) is to make sure the trace flushes before the process completes. My concern would be that this somehow overlaps with the global `at_exit` hook that I think would run in the fork anyways.

Reply:
This looks like a bug... Looking at the Qless sources:

https://github.com/seomoz/qless/blob/7ac9f3a42b8fac9167cda5ccf117b967c7a25e65/lib/qless/worker/forking.rb#L238

...the `#after_fork` method seems to be called after forking a worker process, so it seems off that we'd want to shutdown tracing for that worker.